Assets/Scripts/Environment/InteractableObject.cs

```diff
@@ -43,6 +44,10 @@ public virtual void OnTriggerEnter(Collider other)
     public virtual void OnTriggerExit(Collider other)
     {
         Debug.Log("On trigger exit Interactable. " + other.name);
-        SetPlayerIsColliding(other);
+        if(other.CompareTag(PlayerConstants.PlayerTag) == true)
+        {
+            IsPlayerColliding = false;
+        }
+        Debug.Log("IsPlayerColliding = " + IsPlayerColliding);
     }

```

Comment:
I modified this function to make sure that it is the player exiting the collider that sets IsPlayerColliding to false. Before that, any collider that exited set playerIsColliding to false.